x-pack/plugins/session_view/public/components/SessionViewDetailPanel/index.tsx

```diff
@@ -0,0 +1,148 @@
+  content: ReactNode;
+}
+
+const DETAIL_PANEL_COMMAND = i18n.translate('xpack.sessionView.detailPanel.detailPanelCommand', {
+  defaultMessage: 'Command detail',
+});
+
+const DETAIL_PANEL_SESSION = i18n.translate('xpack.sessionView.detailPanel.detailPanelSession', {
+  defaultMessage: 'Session detail',
+});
+
+const DETAIL_PANEL_SERVER = i18n.translate('xpack.sessionView.detailPanel.detailPanelServer', {
+  defaultMessage: 'Server detail',
+});
+
+const DETAIL_PANEL_ALERT = i18n.translate('xpack.sessionView.detailPanel.detailPanelAlert', {
+  defaultMessage: 'Alert detail',
+});
+
+/**
+ * Detail panel in the session view.
+ */
+export const SessionViewDetailPanel = ({
+  isDetailMounted,
+  height,
+  selectedProcess,
+  setIsDetailOpen,
+  session,
+}: SessionViewDetailPanelDeps) => {
+  const [selectedDetailTab, setSelectedDetailTab] = useState<string | number>('');
+  const [processEventsTabs, setProcessEventsTabs] = useState<ProcessEventTabData[]>([]);
+
+  const styles = useStyles({ height });
+
+  useEffect(() => {
+    const selectedProcessEvents = (selectedProcess?.events || []).map((processEvent, idx) => ({
+      id: `${processEvent?.event.action}-${idx + 1}` || `event-${idx + 1}`,
+      name: `${processEvent?.event.action}-${idx + 1}` || `event-${idx + 1}`,
+      content: (
+        <div>
+          <MonacoEditor
+            height={400}
+            language="json"
+            options={{
+              lineNumbers: 'on',
+              readOnly: true,
+            }}
+            value={JSON.stringify(processEvent || {}, null, 4)}
+          />
+          <EuiSpacer size="xxl" />
+        </div>
+      ),
+    }));
+
+    setProcessEventsTabs(selectedProcessEvents);
+    setSelectedDetailTab(selectedProcessEvents?.[0].id || '');
+  }, [selectedProcess]);
+
+  const handleAnimationEnd = () => {
+    if (!isDetailMounted) {
+      setIsDetailOpen(false);
+    }
+  };
+
+  const renderSelectedProcessEvents = () => {
+    if (selectedProcess) {
+      return (
+        <div>
+          <EuiTitle size="s">
+            <span>{DETAIL_PANEL_COMMAND}</span>
+          </EuiTitle>
+          <EuiSpacer />
+          <EuiTabs>
+            {processEventsTabs.map((tab, idx) => (
+              <EuiTab
+                key={idx}
+                onClick={() => setSelectedDetailTab(tab.id)}
+                isSelected={tab.id === selectedDetailTab}
+              >
+                {tab.name}
+              </EuiTab>
+            ))}
+          </EuiTabs>
+          <EuiSpacer size="xxl" />
+          {processEventsTabs.find((tab) => tab.id === selectedDetailTab)?.content}
+        </div>
+      );
+    }
+  };
+
+  return (
+    <EuiSplitPanel.Inner
+      paddingSize="s"
+      color="plain"
+      css={isDetailMounted ? styles.detailPanelIn : styles.detailPanelOut}
+      onAnimationEnd={handleAnimationEnd}
+    >
+      {renderSelectedProcessEvents()}
+      <EuiTitle size="s">
+        <span>{DETAIL_PANEL_SESSION}</span>
```

Comment:
Do we need this if we are just showing raw json for each of the process events? Not sure if giving process.session it's own section is worth doing at this point. I think as long as it's allowing us to explore each of the events for the process, that should be enough to help design make some decisions about how to move forward with UX.

Reply:
i will remove this